### ISSUE TITLE
Consider promoting platform.system over sys.platform and os.name

### ISSUE THREAD

**Issue by NeilGirdhar** [astral-sh/ruff#5622]
See [this explanation](https://stackoverflow.com/questions/1854/how-to-identify-which-os-python-is-running-on/58071295#58071295).

**Comment by real-or-random** [comment]
I disagree. Different methods have different purposes. As stated in the linked post, if you only care whether you're on posix or not, `os.name == "posix"` is the right thing to do.

**Comment by NeilGirdhar** [comment]
> I disagree. Different methods have different purposes. As stated in the linked post, if you only care whether you're on posix or not, `os.name == "posix"` is the right thing to do.

Why not do `platform.system() in {'Linux', 'Darwin'}`?  It's easier to understand, it's easier to alter if what you're checking changes, and it doesn't depend on what future Python versions do with "OS specific modules".

**Comment by MichaReiser** [comment]
`sys.platforms` is used and [understood by type checkers](https://typing.python.org/en/latest/spec/directives.html#version-and-platform-checking) whereas `os.platform` isn't (but some type checkers do support it, e.g. Pyright). 

A use case where you want to use `sys.platform` is when checking for specific APIs, e.g. https://docs.python.org/3/library/sys.html#sys.activate_stack_trampoline which is only available on linux and not all posix systems.

**Comment by NeilGirdhar** [comment]
> `sys.platforms` is used and [understood by type checkers](https://typing.python.org/en/latest/spec/directives.html#version-and-platform-checking) whereas `os.platform` isn't (but some type checkers do support it, e.g. Pyright).
> 
> A use case where you want to use `sys.platform` is when checking for specific APIs, e.g. https://docs.python.org/3/library/sys.html#sys.activate_stack_trampoline which is only available on linux and not all posix systems.

How would you check that?   Wouldn't `platform.system() == 'Linux'` be what you want?  As opposed to `sys.platform.startswith('linux')`, which seems much worse.

**Comment by MichaReiser** [comment]
I'd use `sys.platform == "linux"`? `startswith` shouldn't be necessary anymore because of 

> Changed in version 3.3: On Linux, [sys.platform](https://docs.python.org/3/library/sys.html#sys.platform) doesn’t contain the major version anymore. It is always 'linux', instead of 'linux2' or 'linux3'.

**Comment by NeilGirdhar** [comment]
Okay.  I'm not sure if it's possible, but I think it would be good to come up with canonical ways of accomplishing various tasks and have Ruff nudge users towards these ways.  If `sys.platform == 'linux'` is better than `platform.system() == 'Linux'`, (or vice versa), it would be good to nudge everyone to write the same idea in the same way.
